stream-listener-json/src/main/java/demo/SampleSource.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package demo;
+
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.Output;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.annotation.InboundChannelAdapter;
+import org.springframework.integration.annotation.Poller;
+import org.springframework.integration.core.MessageSource;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+
+/**
+ * @author Ilayaperumal Gopinathan
+ */
+@EnableBinding(SampleSource.Source.class)
+public class SampleSource {
+
+	@Bean
+	@InboundChannelAdapter(value = Source.SAMPLE, poller = @Poller(fixedDelay = "1000", maxMessagesPerPoll = "1"))
+	public MessageSource<String> timerMessageSource() {
+		return new MessageSource<String>() {
+			public Message<String> receive() {
+				System.out.println("******************");
+				System.out.println("At the Source");
+				System.out.println("******************");
+				String message = "{\"value\":\"test\"}";
+				System.out.println("Sending JSON message: " + message);
+				System.out.println("******************");
+				return MessageBuilder.withPayload(message).setHeader("contentType", "application/json").build();
+			}
+		};
+	}
+
+	public interface Source {
```

Comment:
Maybe use spring.cloud.stream.bindings.output.contentType=application/json instead and simplify the whole method to get rid of MessageSource and just return string.